microsoft-edge/webview2/get-started/hololens2.md

```diff
@@ -0,0 +1,627 @@
+
+`IWithPostMessage` also defines the `PostMessage()` method, which can be used to send a message from the Unity app to the webview.
+
+Here is an example of how to use these events and methods for working with the webview:
+
+```C#
+using Microsoft.MixedReality.WebView;
+
+public class WebViewExample : MonoBehaviour, IWithPostMessage
+{
+    private void Start()
+    {
+        var webViewComponent = gameObject.GetComponent<WebView>();
+
+        if (Url != string.Empty)
+        {
+            Debug.Log("Loading URL: " + Url);
+            webViewComponent.Load(new Uri(Url));
+        }
+
+        webViewComponent.GetWebViewWhenReady((IWebView webView) =>
+        {
+            // Add event listeners for WebView2 events
+            ((IWithPostMessage)webView).MessageReceived += OnMessageReceived;
+        });
+    }
+
+    // Handler for WebView2 OnPostMessage event
+    void OnMessageReceived(string message)
+    {
+        Debug.Log("WebView2 message received: " + message);
+    }
+}
+```
+
+<!-- ====================================================================== -->
+## Additional considerations for WebView2 on HoloLens 2
+
+
+<!-- ------------------------------ -->
+#### Limitations and known issues
+
+When developing a HoloLens 2 Unity app with WebView2, be aware of some limitations and known issues:
+
+* **Pop-ups**: Pop-ups don't work well within WebView2 inside Unity apps on HoloLens 2, but they work fine in 2D XAML apps on the device. Avoid pop-ups and use alternative techniques or UI designs, such as custom pop-up-like elements within the WebView using HTML, CSS, and JavaScript.
+
+* **New windows**: WebView2 instances on HoloLens 2 navigate within the same window by default, unlike on Desktop. Follow this default behavior for a better user experience. Additionally, the DevTools window cannot be launched.
+
+* **Enterprise authentication**: Automatic Single Sign-On (SSO) leveraging OS-level tokens is currently not supported in WebView2 on HoloLens 2. Users can still sign in by providing credentials, except for cases requiring device-level authentication. Cookie storage works as expected.
+
+* **User interactions**: Unlike native HoloLens 2 slates, WebView2 is best interacted with using far-interaction hand rays. Touch-to-swipe and scroll interactions might not be supported.
+
+* **Performance**: Complex websites with heavy use of JavaScript or advanced rendering may impact system performance or the host application's framerate. For general performance-related limitations and recommendations, see [Understanding performance for mixed reality](/windows/mixed-reality/develop/advanced-concepts/understanding-performance-for-mixed-reality) in the Mixed Reality documentation. Also see [Performance optimization](#performance-optimization), below.
+
+
+<!-- ------------------------------ -->
+#### Navigation
+
+In Step 6, we touched on some navigation methods. In this section, we'll expand on what we learned.
+
+
+<!-- ---------- -->
```

Comment:
i think this is the "APIs" section Rob suggests moving to a new sep. article

Reply:
@RobertVS-MS - at some point, this page will need to link to the forthcoming API Ref, in the other repo.  Maybe I will add a hidden link here and in See Also at bottom.

Reply:
I'd like to keep them. Reeza suggested we can add links to the API documention after it goes live.

Reply:
can close this thread